### ISSUE TITLE
Network tutorial

### ISSUE THREAD

**Issue by everett-toews** [getcarina/getcarina.com#686]

**Comment by rosskukulinski** [comment]
@everett-toews something to touch on: what's the process for getting access to these fancy new networks?  Do existing clusters need to be destroyed and new ones created?  Is there an in-place upgrade process?

**Comment by rgbkrk** [comment]
You'll need to create a new cluster.

**Comment by everett-toews** [comment]
Blog post on it here: https://getcarina.com/blog/overlay-networks/

**Comment by everett-toews** [comment]
From @ktbartholomew 

> Question: can you access ServiceNet from an overlay network?
> They're both in 10.0.... but the netmask is pretty small there
> Not sure what the full IP range of ServiceNet is, but we may want to recommend explicitly setting the subnet just to be safe

This will need a bit of investigation.

**Comment by everett-toews** [comment]
See [this comment](https://github.com/everett-toews/jupyterhub-on-docker-swarm/issues/16#issuecomment-185869665) on security. The tutorial should have a section dedicated to security.

**Comment by everett-toews** [comment]
Here's the pathological case of a docker network IP colliding with a ServiceNet IP on Carina

https://gist.github.com/everett-toews/68e4677cf637b976a78a

tl;dr If you have an IP collision, you can't reach that container by name or IP outside of the segment. No real surprise there.

The ServiceNet CIDR is 10.176.224.0/19 which is a far cry from from the docker default of 10.0.0.0/24 (incrementing by 10.0.1.0/24 as new networks are created). You'd have to create thousands of networks with the default settings before you collided with the ServiceNet CIDR. Asking users to include `--subnet` in their `docker network create` commands is a cognitive burden and one more decision point to frustrate them. Therefore we're recommending not using `--subnet` but documenting the caveat that the ServiceNet CIDR is 10.176.224.0/19 in this networking tutorial and a troubleshooting tip.

**Comment by carolynvs** [comment]
@everett-toews I assume that this will replace https://getcarina.com/docs/concepts/docker-networking-basics/? Do you want me to do anything right now to that umbrella article, or the two child articles on ambassador pattern and legacy linking as part of #714 ? Seems like they should just be deleted once you rework the main article?

**Comment by everett-toews** [comment]
@carolynvs You're right. All 3 can be deleted. The ambassador pattern and legacy linking articles make no sense anymore and the concept umbrella article is entirely about those 2 articles so it doesn't make sense either.

I'm writing this as a tutorial specifically about overlay networks. If we want to expand on networking in general then, at that time, it would make sense to reintroduce a concept article on networking.

**Comment by carolynvs** [comment]
@everett-toews Do you know when the overlay tutorial is going out? I can remove the networking tutorials, but a few other pages are linking to the existing networking basics tutorial. If it's coming out soon, I can just update them to point to the upcoming overlay tutorial URL.

**Comment by everett-toews** [comment]
@carolynvs I'll have a PR out today. Hopefully it'll be published on Tuesday or Wednesday.
